### PR TITLE
mgmt, local generation for Storage - 2023-05-01

### PR DESCRIPTION
**Pull request by v-hongli1**
relate to [issue#5259](https://github.com/Azure/sdk-release-request/issues/5259)

Manual code:
sdk/resourcemanager/azure-resourcemanager-storage/src/test/java/com/azure/resourcemanager/storage/StorageAccountOperationsTests.java

Manual code reason：
The default value of the parameter `cross-tenant replication` of `Storage account` has been changed from `true` to `false`.

# Description

Please add an informative description that covers that changes made by the pull request and link all relevant issues.

If an SDK is being regenerated based on a new swagger spec, a link to the pull request containing these swagger spec changes has been included above.

# All SDK Contribution checklist:
- [ ] **The pull request does not introduce [breaking changes]**
- [ ] **CHANGELOG is updated for new features, bug fixes or other significant changes.**
- [ ] **I have read the [contribution guidelines](https://github.com/Azure/azure-sdk-for-java/blob/main/CONTRIBUTING.md).**

## [General Guidelines and Best Practices](https://github.com/Azure/azure-sdk-for-java/blob/main/CONTRIBUTING.md#developer-guide)
- [ ] Title of the pull request is clear and informative.
- [ ] There are a small number of commits, each of which have an informative message. This means that previously merged commits do not appear in the history of the PR. For more information on cleaning up the commits in your PR, [see this page](https://github.com/Azure/azure-powershell/blob/master/documentation/development-docs/cleaning-up-commits.md).

### [Testing Guidelines](https://github.com/Azure/azure-sdk-for-java/blob/main/CONTRIBUTING.md#building-and-unit-testing)
- [ ] Pull request includes test coverage for the included changes.

### CODE REVIEW

File: sdk/resourcemanager/azure-resourcemanager-storage/src/test/java/com/azure/resourcemanager/storage/StorageAccountOperationsTests.java [hunk 1]
```diff
@@ -221,6 +221,12 @@ public void canDisallowCrossTenantReplicationOnStorageAccount() {
                 .withSku(StorageAccountSkuType.STANDARD_LRS)
                 .create();
 
+        Assertions.assertFalse(storageAccount.isAllowCrossTenantReplication());
+
+        storageAccount.update()
+            .allowCrossTenantReplication()
+            .apply();
+
```
Comment (XiaofeiCao) [inline sdk/resourcemanager/azure-resourcemanager-storage/src/test/java/com/azure/resourcemanager/storage/StorageAccountOperationsTests.java:229]: Thanks. Should we confirm with service about this behavioral change? Guess if it's expected, we should put that in breaking change section.

Comment (v-hongli1) [reply sdk/resourcemanager/azure-resourcemanager-storage/src/test/java/com/azure/resourcemanager/storage/StorageAccountOperationsTests.java:229]: When creating a Storage Account on the portal, the default value of this parameter is `false`.
image

If it is a breaking change, the `allowCrossTenantReplication` interface must be added in `Create` mode.

Comment (weidongxu-microsoft) [reply sdk/resourcemanager/azure-resourcemanager-storage/src/test/java/com/azure/resourcemanager/storage/StorageAccountOperationsTests.java:229]: It appears being changed due to security reason. Please document it in breaking changes
https://learn.microsoft.com/en-us/rest/api/storagerp/storage-accounts/create?view=rest-storagerp-2023-01-01

Comment (weidongxu-microsoft) [reply sdk/resourcemanager/azure-resourcemanager-storage/src/test/java/com/azure/resourcemanager/storage/StorageAccountOperationsTests.java:229]: You can merge this PR first, and immediately make a follow-up PR to add `allowCrossTenantReplication` API to create flow.
https://github.com/azure/azure-sdk-for-java/blob/main/sdk/resourcemanager/azure-resourcemanager-storage/src/main/java/com/azure/resourcemanager/storage/models/StorageAccount.java#L675

**Review by weidongxu-microsoft — APPROVED** [review]

File: sdk/resourcemanager/azure-resourcemanager-storage/src/test/java/com/azure/resourcemanager/storage/StorageAccountOperationsTests.java (hunk 1 above)
Comment (v-hongli1) [reply sdk/resourcemanager/azure-resourcemanager-storage/src/test/java/com/azure/resourcemanager/storage/StorageAccountOperationsTests.java:229]: So do we need to fix `Create` mode?

Comment (weidongxu-microsoft) [reply sdk/resourcemanager/azure-resourcemanager-storage/src/test/java/com/azure/resourcemanager/storage/StorageAccountOperationsTests.java:229]: Yes, in a follow-up PR.

**Review by XiaofeiCao — APPROVED** [review]

**Pull request merged**